src/pcre2_compile.c

```diff
@@ -4233,7 +4233,7 @@ while (ptr < ptrend)
                 }
               if (*ptr == CHAR_P)
                 {
-                *xoptset |= PCRE2_EXTRA_ASCII_POSIX;
+                *xoptset |= (PCRE2_EXTRA_ASCII_POSIX|PCRE2_EXTRA_ASCII_DIGIT);
```

Comment:
I don't see why you want to set both flags here, since ASCII_POSIX affects all POSIX classes (thanks for adding emphasis to the docs, btw).

Reply:
because without it, (?aP) was inconsistent as shown by:

```
  re> /(?-aP:[[:digit:]])[[:digit:]]/B,utf,ucp,ascii_digit
------------------------------------------------------------------
        Bra
        Bra
        [0-9]
        Ket
        [0-9]
        Ket
        End
------------------------------------------------------------------
```